patches/server/0031-Added-explosionproof-items.patch

```diff
@@ -0,0 +1,55 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Hermelijn15 <j.s.dekker-1@student.utwente.nl>
+Date: Thu, 16 Nov 2023 11:40:18 +0100
+Subject: [PATCH] Added explosionproof items
+
+
+diff --git a/src/main/java/me/samsuik/sakura/configuration/WorldConfiguration.java b/src/main/java/me/samsuik/sakura/configuration/WorldConfiguration.java
+index deaee9363553df0ab5aff453c278a51987d4ebb1..24e0f04bdbdeaa36fb17035e506401b6d11ff666 100644
+--- a/src/main/java/me/samsuik/sakura/configuration/WorldConfiguration.java
++++ b/src/main/java/me/samsuik/sakura/configuration/WorldConfiguration.java
+@@ -10,14 +10,18 @@ import me.samsuik.sakura.entity.merge.MergeLevel;
+ import me.samsuik.sakura.explosion.durable.DurableMaterial;
+ import net.minecraft.Util;
+ import net.minecraft.resources.ResourceLocation;
++import net.minecraft.world.item.Item;
+ import net.minecraft.world.level.block.Block;
+ import net.minecraft.world.level.block.Blocks;
+ import org.slf4j.Logger;
+ import org.spongepowered.configurate.objectmapping.meta.Comment;
+ import org.spongepowered.configurate.objectmapping.meta.Setting;
+ 
++import java.util.List;
+ import java.util.Map;
+ 
++import static net.minecraft.world.item.Items.SPAWNER;
++
+ @SuppressWarnings({"FieldCanBeLocal", "FieldMayBeFinal", "NotNullFieldNotInitialized", "InnerClassMayBeStatic", "RedundantSuppression"})
+ public class WorldConfiguration extends ConfigurationPart {
+ 
+@@ -104,4 +108,11 @@ public class WorldConfiguration extends ConfigurationPart {
+         }
+     }
+ 
++    public Items items;
++    public class Items extends ConfigurationPart {
++        public ExplosionProof explosionProof = new ExplosionProof();
++        public class ExplosionProof extends ConfigurationPart {
++            public List<Item> explosionResistentItems = List.of(SPAWNER);
++        }
++    }
+ }
+diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
+index 50259042ebafe520342bdb1a4b7e6b9138b8acbd..5c7c6498f3a2ac4846568760b5b09b3858d419d1 100644
+--- a/src/main/java/net/minecraft/world/entity/Entity.java
++++ b/src/main/java/net/minecraft/world/entity/Entity.java
+@@ -4412,6 +4412,9 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource {
+     }
+ 
+     public boolean ignoreExplosion() {
++        if(this instanceof ItemEntity && level.sakuraConfig().items.explosionProof.explosionResistentItems.contains(((ItemEntity) this).getItem().getItem())){
```

Comment:
Missing // Sakura start and // Sakura end comments and this logic should be in ItemEntity by overriding the ignoreExplosion method.